File: java/command/Light.java
package command;

public class Light {

    private String name;

    public Light(String name) {
        this.name = name;
    }

    public void turnOn() {
        System.out.println("Turning on light....");
    }

    public void turnOff() {
        System.out.println("Turning off light....");
    }

    public String getName() {
        return name;
    }
}

File: typescript/patterns/command/light.ts
class Light {
  private name: string;

  constructor(name: string) {
    this.name = name;
  }

  turnOn(): void {
    console.log(`Turning on Light ${this.name}...`);
  }

  turnOff(): void {
    console.log(`Turning off Light ${this.name}...`);
  }

  getName(): string {
    return this.name;
  }
}
